android/src/main/java/io/radar/react/RNRadarModule.java

```diff
@@ -31,23 +40,38 @@
 import io.radar.sdk.model.RadarPlace;
 import io.radar.sdk.model.RadarRouteMatrix;
 import io.radar.sdk.model.RadarRoutes;
+import io.radar.sdk.model.RadarTrip;
 import io.radar.sdk.model.RadarUser;
 
-import org.json.JSONException;
-import org.json.JSONObject;
-
-import java.util.EnumSet;
-
+/**
+ * React-Native module to interface with the Radar Android SDK
+ *
+ * @see <a href="https://radar.io/documentation/sdk/react-native">
+ * Radar React Native SDK Reference
+ * </a>
+ */
+@SuppressWarnings("unused")
 public class RNRadarModule extends ReactContextBaseJavaModule implements PermissionListener {
 
     private static final String TAG = "RNRadarModule";
-    private static final int PERMISSIONS_REQUEST_CODE = 20160525; // random request code (Radar's birthday!)
+    // random request code (Radar's birthday!)
+    private static final int PERMISSIONS_REQUEST_CODE = 20160525;
     private Promise mPermissionsRequestPromise;
 
     public RNRadarModule(ReactApplicationContext reactContext) {
         super(reactContext);
+        String publishableKey = reactContext.getString(R.string.radar_publishableKey);
```

Comment:
Nice new pattern.  So I guess now there are 3 ways to initialize the sdk?

1) configure `radar_publishableKey` in `strings.xml`
2) Call `Radar.initialize()` manually
3) Use the new `RNRadar` helper

Wondering if we want to have 3 different ways to do it, or if that's too many choices.  Could be fine as long as we document it and write a good migration guide.  1) Seems like far and away the best way for most apps to do it. cc @brettguenther @nickpatrick who have more experience with customer set ups here.

Reply:
cc @jsani-radar

Reply:
Out of the 3, my gut says to support 1 and 2 - I'm not sure why someone would use a new helper (#3) when this PR gives the ability for devs to paste a string in an XML file. `Radar.initialize()` is the legacy way of doing things, so I think we should keep that to continue supporting existing apps and avoid a breaking change

Reply:
The rationale was that the Radar.initialize method does not require any receiver, which makes it so that the hooks into events for RN are not received - so when an existing customer updates to use 3.2, this might easily be missed. Maybe instead of adding RNRadar, it makes sense to do a new minor release number with good docs?

Reply:
ah gotcha, looks like we landed on that approach for [Capacitor]([https://github.com/radarlabs/capacitor-radar/pull/25#discussion_r764265352])

Reply:
@jsani-radar they will still be able to use 2. I can add back `RNRadar`, giving all 3 options - or we can just have 1 & 2, and have some good documentation for migrating.